apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.js

```diff
@@ -3,11 +3,113 @@
 window.PLOrderBlocks = function (uuid, options) {
   const TABWIDTH = 50; // defines how many px the answer block is indented by, when the student
   // drags and indents a block
-  let maxIndent = options.maxIndent; // defines the maximum number of times an answer block can be indented
-  let enableIndentation = options.enableIndentation;
+  const maxIndent = options.maxIndent; // defines the maximum number of times an answer block can be indented
+  const enableIndentation = options.enableIndentation;
 
-  let optionsElementId = '#order-blocks-options-' + uuid;
-  let dropzoneElementId = '#order-blocks-dropzone-' + uuid;
+  const optionsElementId = '#order-blocks-options-' + uuid;
+  const dropzoneElementId = '#order-blocks-dropzone-' + uuid;
+  const fullContainer = document.querySelector('.pl-order-blocks-question-' + uuid);
+
+  function initializeKeyboardHandling() {
+    const blocks = fullContainer.querySelectorAll('.pl-order-block');
+    blocks.forEach((block) => initializeBlockEvents(block));
+  }
+
+  function inDropzone(block) {
+    const parentArea = block.closest('.pl-order-blocks-connected-sortable');
+    return parentArea.classList.contains('dropzone');
+  }
+
+  function getIndentation(block) {
+    return Math.round(parseInt(block.style.marginLeft.replace('px', '') / TABWIDTH));
+  }
+
+  function setIndentation(block, indentation) {
+    if (indentation >= 0 && indentation <= maxIndent) {
+      block.style.marginLeft = indentation * TABWIDTH + 'px';
+    }
+  }
```

Comment:
I think retrieving the indentation from the margin may be a bit volatile. Have you considered using a [data attribute](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) with the indentation level for a block?

Reply:
I agree that seems like a good idea. This isn't really new in the codebase though, this is just rearranging some code that was already there, so I'd rather do that in a separate PR.